### PR TITLE
Get default configurations for Model FineTuning

### PR DESCRIPTION
**Pull request by VipulMascarenhas**
### Description
This PR is built on top of changes made in [PR-827](https://github.com/oracle/accelerated-data-science/pull/827). 
We add a new API that returns the default params used for a fine tuning job. These params are read from ft_config.json but only those that are available in `AquaFineTuningParams` dataclass will be made available to the user. 

#### Request
```
http://localhost:8888/aqua/finetuning/ocid1.datasciencemodel.oc1.iad.<ocid>/params
```
#### Response
```
{
    "data": [
        "--batch_size 1",
        "--sequence_len 2048",
        "--sample_packing true",
        "--pad_to_sequence_len true",
        "--learning_rate 0.0002",
        "--lora_r 32",
        "--lora_alpha 16",
        "--lora_dropout 0.05",
        "--lora_target_linear true",
        "--lora_target_modules q_proj,k_proj"
    ]
}
```
If ft_config.json is not available for the given model is, then the response will be:
```
{
    "data": []
}

```

### Tests
```
> python -m pytest -q tests/unitary/with_extras/aqua/test_finetuning.py
========================================= test session starts ==========================================
platform darwin -- Python 3.8.18, pytest-7.4.0, pluggy-1.0.0
rootdir: /Users/user/workspace/git/accelerated-data-science
configfile: pytest.ini
plugins: Faker-24.9.0, anyio-4.2.0
collected 4 items

tests/unitary/with_extras/aqua/test_finetuning.py ....                                           [100%]

========================================== 4 passed in 3.67s ===========================================


> python -m pytest -q tests/unitary/with_extras/aqua/test_finetuning_handler.py
========================================= test session starts ==========================================
platform darwin -- Python 3.8.18, pytest-7.4.0, pluggy-1.0.0
rootdir: /Users/user/workspace/git/accelerated-data-science
configfile: pytest.ini
plugins: Faker-24.9.0, anyio-4.2.0
collected 4 items

tests/unitary/with_extras/aqua/test_finetuning_handler.py ....                                   [100%]

========================================== 4 passed in 2.98s ===========================================
```

### CODE REVIEW

File: ads/aqua/extension/finetune_handler.py [hunk 1]
```diff
@@ -59,7 +59,20 @@ def get_finetuning_config(self, model_id):
         return self.finish(AquaFineTuningApp().get_finetuning_config(model_id=model_id))
 
 
+class AquaFineTuneParamsHandler(AquaAPIhandler):
+    """Handler for Aqua finetuning params REST APIs."""
+
+    @handle_exceptions
+    def get(self, model_id):
+        """Handle GET request."""
+        model_id = model_id.split("/")[0]
```
Comment (mayoor) [inline ads/aqua/extension/finetune_handler.py:68]: Can you please add a comment on why there is a split?

File: ads/aqua/finetune.py [hunk 2]
```diff
@@ -650,3 +654,36 @@ def get_finetuning_config(self, model_id: str) -> Dict:
...
+        default_params = []
+        finetuning_config = self.get_finetuning_config(model_id)
+        config_parameters = finetuning_config.get("configuration", UNKNOWN_DICT)
+        dataclass_fields = {field.name for field in fields(AquaFineTuningParams)}
+        for name, value in config_parameters.items():
+            if name == "micro_batch_size":
+                name = "batch_size"
+            if name == "lora_target_modules":
+                value = ",".join(str(k) for k in value)
+            if name in dataclass_fields:
```
Comment (mayoor) [inline ads/aqua/finetune.py:686]: We are already validating somewhere that all the hyperparameter attibute is subset of this data class right?

Comment (mayoor) [reply ads/aqua/finetune.py:686]: Nvm. I confused this with create FT.

File: ads/aqua/extension/finetune_handler.py (hunk 1 above)
Comment (VipulMascarenhas) [reply ads/aqua/extension/finetune_handler.py]: fixed the handler regex, now we don't need to split. This is present throughout in aqua handlers, will add it to refactor task to update the regex.

File: ads/aqua/finetune.py [hunk 3]
```diff
@@ -650,3 +654,36 @@ def get_finetuning_config(self, model_id: str) -> Dict:
...
+
+        Parameters
+        ----------
+        model_id: str
+            The OCID of the Aqua model.
+
+        Returns
+        -------
+        List[str]:
+            List of parameters from the loaded from finetuning config json file. If not available, then an empty list
```
Comment (mrDzurb) [inline ads/aqua/finetune.py:674]: NIT: ```List of parameters  loaded from finetuning config json file. If not available, then an empty list```?

**Review by mrDzurb — APPROVED** [review]
LGTM

**Pull request merged**